src/components/select/select-textbox/select-textbox.component.js

```diff
@@ -0,0 +1,120 @@
+import Textbox from '../../../__experimental__/components/textbox';
+import OptionsHelper from '../../../utils/helpers/options-helper/options-helper';
+
+const SelectTextbox = ({
+  value,
+  disabled,
+  readOnly,
+  placeholder,
+  onClick,
+  onFocus,
+  onBlur,
+  selectedValue,
+  ...restProps
+}) => {
+  const defaultPlaceholder = I18n.t('select.placeholder', {
+    defaultValue: 'Please Select...'
+  });
+
+  function handleTextboxClick(event) {
+    if (disabled || readOnly) {
+      return;
+    }
+
+    onClick(event);
+  }
+
+  function handleTextboxFocus(event) {
+    if (disabled || readOnly) {
+      return;
+    }
+
+    if (onFocus) {
+      onFocus(event);
+    }
+  }
+
+  function handleTextboxBlur(event) {
+    if (onBlur) {
+      onBlur(event);
+    }
+  }
+
+  function getTextboxProps() {
+    return {
+      placeholder: placeholder || defaultPlaceholder,
+      disabled,
+      readOnly,
+      onClick: handleTextboxClick,
+      onFocus: handleTextboxFocus,
+      onBlur: handleTextboxBlur,
+      ...restProps
+    };
+  }
+
+  return (
+    <Textbox
+      data-element='select-input'
+      inputIcon='dropdown'
+      autoComplete='off'
+      value={ selectedValue }
+      { ...getTextboxProps() }
+    />
+  );
+};
+
+const formInputPropTypes = {
+  /** Id attribute of the input element */
+  id: PropTypes.string,
+  /** Name attribute of the input element */
+  name: PropTypes.string,
+  /** If true the Component will be read-only */
+  readOnly: PropTypes.bool,
+  /** If true the Component will be disabled */
+  disabled: PropTypes.bool,
+  /** If true the Component will be focused when rendered */
+  autoFocus: PropTypes.bool,
+  /** A callback to retrieve the input reference */
+  inputRef: PropTypes.func,
+  /** Label */
+  label: PropTypes.string,
+  /** Text applied to label help tooltip */
+  labelHelp: PropTypes.string,
+  /** When true, label is placed in line with an input */
+  labelInline: PropTypes.bool,
+  /** Width of a label in percentage. Works only when labelInline is true */
+  labelWidth: PropTypes.number,
+  /** Width of an input in percentage. Works only when labelInline is true */
+  inputWidth: PropTypes.number,
+  /** Size of an input */
+  size: PropTypes.oneOf(OptionsHelper.sizesRestricted),
+  /** Placeholder string to be displayed in input */
+  placeholder: PropTypes.string,
+  /** A custom callback for when changes occur */
+  onChange: PropTypes.func,
+  /** Callback function for when the Select Textbox is clicked. */
+  onClick: PropTypes.func,
+  /** Callback function for when the Select Textbox is focused. */
+  onFocus: PropTypes.func,
+  /** Callback function for when the Select Textbox loses it's focus. */
+  onBlur: PropTypes.func,
+  /** Callback function for when the key is pressed when focused on Select Textbox. */
+  onKeyDown: PropTypes.func
+};
+
+SelectTextbox.propTypes = {
+  ...formInputPropTypes,
+  /** @private Value to be displayed in the Textbox */
```

Comment:
`@private` correctly tells developer that this prop is internal, but it doesn't hide it in storybook.
We also need to use `@internal`.

Reply:
you mean `@ignore`?

Reply:
Yes, sorry for the confusion.